test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Tests.ps1

```diff
@@ -0,0 +1,194 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+$script:modulePathStack = [System.Collections.Generic.Stack[string]]::new()
+function Push-ModulePath
```

Comment:
Seems overkill since it doesn't appear the tests need to push more than once before a pop.  Existing tests simply store `currentPSModulePath` in `Before` and restore in `After`